src/olympia/addons/views.py

```diff
@@ -859,3 +859,26 @@ def finalize_response(self, request, response, *args, **kwargs):
             request, response, *args, **kwargs)
         patch_cache_control(response, max_age=60 * 60 * 6)
         return response
+
+
+class LanguageToolsView(ListAPIView):
+    authentication_classes = []
+    pagination_class = None
+    permission_classes = []
+    serializer_class = LanguageToolsSerializer
+
+    def get_queryset(self):
+        try:
+            application_id = AddonAppFilterParam(self.request).get_value()
+        except ValueError:
+            raise ParseError('Invalid app parameter.')
+
+        types = (amo.ADDON_DICT, amo.ADDON_LPAPP)
+        return Addon.objects.public().filter(
```

Comment:
I played with this in a mysql shell and the current sort, through undefined (it's not necessarily add-on id), seems to perform better - essentially mysql is picking rows in the index order, not having to sort them.

Since the client will have to re-order everything anyway, I'm going to keep this and document it.